From: Philip Withnall <pwithnall@endlessos.org>
Date: Mon, 6 Mar 2023 14:38:22 +0000
Subject: glocalfileinfo: Set is-backup attribute on Windows

Previously it was only being set on non-Windows platforms. For
consistency, always set it on Windows too.

Signed-off-by: Philip Withnall <pwithnall@endlessos.org>

Helps: #2934
(cherry picked from commit a4e597520a46cbd158c59487e116e603691aaff4)
---
 gio/glocalfileinfo.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gio/glocalfileinfo.c b/gio/glocalfileinfo.c
index 1af3da8..bccad04 100644
--- a/gio/glocalfileinfo.c
+++ b/gio/glocalfileinfo.c
@@ -2051,6 +2051,8 @@ _g_local_file_info_get (const char             *basename,
                                             basename != NULL && basename[strlen (basename) - 1] == '~' &&
                                                 (stat_ok && S_ISREG (_g_stat_mode (&statbuf))));
 #else
+  _g_file_info_set_attribute_boolean_by_id (info, G_FILE_ATTRIBUTE_ID_STANDARD_IS_BACKUP, FALSE);
+
   g_file_info_set_is_hidden (info, (statbuf.attributes & FILE_ATTRIBUTE_HIDDEN));
 
   _g_file_info_set_attribute_boolean_by_id (info, G_FILE_ATTRIBUTE_ID_DOS_IS_ARCHIVE,
